op-succinct-proposer/TUTORIAL.md

```diff
@@ -0,0 +1,109 @@
+# `op-succinct` Tutorial
+
+This tutorial guides you through upgrading an existing OP Stack chain to a ZK-OP Stack chain. It assumes you already have a running OP Stack chain. If not, please follow [Optimism's tutorial](https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup) first.
+
+## Prerequisites
+
+Before starting, ensure you have:
+
+1. Access to the Admin keys for upgrading L1 contracts of the OP Stack chain.
+2. Fault proofs turned off (you should be using `L2OutputOracle`, not `FaultDisputeGameFactory` for output roots).
+3. The original `op-proposer` stopped.
+4. At least 1 ETH in your `PROPOSER` wallet.
+5. All dependencies from the Optimism tutorial installed.
+6. Your L2 geth node running with `--gcmode=archive` and `--state.scheme=hash` flags. You can do this with [ops-anton](https://github.com/anton-rs/ops-anton/blob/main/L2/op-mainnet/op-geth/op-geth.sh).
+
+## Overview
+
+At a high level, the `op-succinct` upgrade focuses on replacing the `op-proposer` component and its associated contracts. This change allows the chain to progress only with ZK-proven blocks, while keeping the other components (`op-geth`, `op-batcher`, and `op-node`) unchanged.
+
+When an OP Stack chain is running, there are 4 main components:
+1. `op-geth` (Sequencer mode): Takes transactions from users and uses them to generate blocks and execute blocks.
+2. `op-batcher`: Batches transactions from users and submits them to the L1.
+3. `op-node`: Reads batch data from L1 and uses it to drive `op-geth` in non-sequencer mode to perform state transitions.
+4. `op-proposer`: Posts an output root to L1 at regular intervals, which captures the L2 state so withdrawals can be processed.
+
+## Step-by-Step Guide
+
+### Step 1: Setup
+
+1. Clone the `op-succinct` repo:
+   ```
+   git clone https://github.com/succinctlabs/op-succinct.git
+   ```
+
+2. Create and fill in the environment files:
+   ```
+   cp .env.example .env
+   cp .env.server.example .env.server
+   ```
+
+3. Fill in the `contracts/zkconfig.json` file with your chain's specific details.
+
+    <details>
+    <summary>Field Info</summary>
+
+    - `startingBlockNumber`: The L1 block number at which the rollup starts. Default should be 0.
```

Comment:
This should be L2 block number. I assume it was my mistake, just noting here to fix before merging.